hack/externalnode/install-vm.sh

```diff
@@ -0,0 +1,164 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+_usage="Usage: $0 [--ns <NameSpace>] [--bin <AntreaAgentSavePath>] [--config <AgentConfigSavePath>] [--kubeconfig-dir <KubeconfigSavePath>] [--help|-h]
+Generate a YAML manifest for Antrea using Helm and print it to stdout.
+        --ns                          NameSpace to be used by the antrea-agent.
+        --bin                         Path of the antrea-agent binary
+        --config                      Path of the antrea-agent configuration file
+        --kubeconfig                  Path of the kubeconfig to access K8s API Server
+        --antrea-kubeconfig           Path of the kubeconfig to access Antrea API Server
+        --help, -h                    Print this message and exit"
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+AGENT_BIN_PATH=""
+CONFIG_PATH=""
+KUBECONFIG=""
+ANTREAKUBECONFIG=""
+AGENT_NAMESPACE=""
+AGENT_LOG_DIR="/var/log/antrea"
+AGENT_CONF_PATH="/etc/antrea"
+
+check_supported_platform() {
+  echo "Checking platform supported"
+  if [[ $(lsb_release -rs) != "20.04" ]]; then
+    echo "Error only Ubuntu 20.04 version is supported"
+    exit 1
+  fi
+}
+
+copy_antrea_agent_files() {
+    if [[ ! -f "$CONFIG_PATH" ]]; then
+      echo "Error $CONFIG_PATH file not found"
+      exit 1
+    fi
+    mkdir -p $AGENT_CONF_PATH
+    echo "Copying $CONFIG_PATH to $AGENT_CONF_PATH"
+    cp $CONFIG_PATH $AGENT_CONF_PATH
+
+    if [[ ! -f "$KUBECONFIG" ]]; then
+      echo "Error $KUBECONFIG file not found"
+      exit 1
+    fi
+
+    echo "Copying $KUBECONFIG to $AGENT_CONF_PATH"
+    cp "$KUBECONFIG" "${AGENT_CONF_PATH}/antrea-agent.kubeconfig"
+
+    if [[ ! -f "$ANTREA_KUBECONFIG" ]]; then
+      echo "Error $ANTREA_KUBECONFIG file not found"
+      exit 1
+    fi
+    echo "Copying $ANTREA_KUBECONFIG to $AGENT_CONF_PATH"
+    cp "$ANTREA_KUBECONFIG" "${AGENT_CONF_PATH}/antrea-agent.antrea.kubeconfig"
+}
+
+update_antrea_agent_conf() {
+  echo "Updating clientConnection and antreaClientConnection"
+  sed -i "s|kubeconfig: |kubeconfig: $AGENT_CONF_PATH/|g" $AGENT_CONF_PATH/antrea-agent.conf
+  echo "Updating externalNodeNamespace to $AGENT_NAMESPACE"
+  sed -i "s|#externalNodeNamespace: default|externalNodeNamespace: $AGENT_NAMESPACE|g" $AGENT_CONF_PATH/antrea-agent.conf
+}
+
+start_antrea_agent_service() {
+    if [[ ! -f "$AGENT_BIN_PATH" ]]; then
+      echo "Error $AGENT_BIN_PATH file not found"
+      exit 1
+    fi
+    mkdir -p $AGENT_LOG_DIR
+cat << EOF > /etc/systemd/system/antrea-agent.service
+Description="antrea-agent as a systemd service"
+After=network.target
+[Service]
+ExecStart=$AGENT_BIN_PATH \
```

Comment:
It seems you forget to set environment variable of `NODE_NAME` in the case VM hostname is different from ExternalNode name

Reply:
I have exposed an optional argument for nodename.